File: Pronic Numbers/Pronic_Number.py
###################################
###     Pronic Numbers          ###
###################################

#Pronic Series Formula 2t(n) = n(n+1) where n>=0
#Example: {0,2,6,12,20,30,42,56,72,90....}

find_val = int(input("Enter the Nth Value : "))
check = 0
ans = 0
iterator = 1
while(check != find_val):
        flag = False
        #Validating Whether the range has any Pronic Number
        for i in range(0,iterator+1):
                if(i*(i+1) == iterator):
                        flag = True
                        break
        if(flag):
                #Updating the answer and incrementing check
                #for outer while loop
                check += 1
                ans = iterator
        iterator += 1
        
print(find_val, "th Pronic Number is ", ans)

File: Pronic Numbers/Pronic_Number.cpp
#include <iostream>
using namespace std;
int main ()
{
  int i, inp, num = 1, c = 0, latest = 0;
  cout<<"Enter the Nth Value: ";
  cin>>inp;
  while (c != inp)
    {
    int flag=0;
     for(i=0;i<=num;i++)
   {
       if(i*(i+1)==num)
       {
           flag=1;
           break;
       }
   }

   if(flag==1)
   {
          c++;
          latest=num;
    }
      num = num + 1;
    }
  cout<<"The "<<inp<<"th Pronic Number is: "<<latest<<"\n";
  return 0;
}